src/Symfony/Component/Routing/Attribute/Route.php

```diff
@@ -199,12 +204,37 @@ public function getPriority(): ?int
         return $this->priority;
     }
 
+    /**
+     * @deprecated since Symfony 7.4, use the {@see setEnvs()} method instead
+     */
     public function setEnv(?string $env): void
     {
-        $this->env = $env;
+        trigger_deprecation('symfony/routing', '7.4', 'The "%s()" method is deprecated, use "setEnvs()" instead.', __METHOD__);
+        $this->env = (array) $env;
     }
 
+    /**
+     * @deprecated since Symfony 7.4, use {@see getEnvs()} method instead
+     */
     public function getEnv(): ?string
+    {
+        trigger_deprecation('symfony/routing', '7.4', 'The "%s()" method is deprecated, use "getEnvs()" instead.', __METHOD__);
+        if (!$this->env) {
+            return null;
+        }
+        if (\count($this->env) > 1) {
+            throw new LogicException(\sprintf('The "env" property has %d environments. Use "getEnvs()" to get all of them.', \count($this->env)));
+        }
+
+        return $this->env[0];
+    }
+
+    public function setEnvs(array $env): void
+    {
+        $this->env = $env;
+    }
```

Comment:
To be consistent with `setSchemes` and `setMethods`, this method should probably accept a string.


```suggestion
    public function setEnvs(array|string $env): void
    {
        $this->env = (array) $env;
    }
```

We keep accepting `null` in the constructor for BC. That could be deprecated. But I'm not sure it's worth it.